81-question-mark-chrysalis/technical_specification.md

```diff
@@ -0,0 +1,202 @@
+# Preliminary Experimental Metadata (EM) Transformation Service (Question Mark Chrysalis)
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://ghga.pages.hzdr.de/internal.ghga.de/main/sops/development/epic_planning/).
+
+## Scope
+
+### Outline:
+
+The goal of this epic is to implement a service for the transformation of Experimental Metadata (EM) from one representation/model to another. 
+This service shall provide functionality around the configurable workflow concept from the `metldata` library to enable these transformations.
+To this goal, it needs to keep track of the transformation workflows, original and derived data and workflow routes in its database.
+
+
+![em-transformation-service overview](./images/em_transformation_service.png)
+
+
+### Terminology
+
+`Experimental Metadata (EM)`: Data describing the experimental process of generating the Research Data archived in GHGA.
+
+`Universal GHGA Model`: Also referred as `EMModel`. A schema that defines how to represent experimental metadata within the GHGA ecosystem.
+
+`EMPack`: Experimental metadata in datapack format. It follows a relation data schema and compatible with one of the Universal GHGA Models. 
+
+`AnnotatedEMPack`: A datapack enriched with an additional information. 
+
+`AnnotatedEMPackCollection`: A MongoDB collection of AnnotatedEMPack objects.
+
+`TransformationWorkflow`: Instructions on how to produce a certain data/model representation from another data/model.
+
+`TransformationWorkflowCollection`: A MongoDB collection of TransformationWorkflow objects.
+
+`WorkflowRoute`: Information on which specific workflow is used to produce data compliant with the output model when presented with data compliant with the input model.
+
+`WorkflowRoutesCollection`: A MongoDB collection of WorkflowRoute objects.
+
+
+### Included/Required:
+
+- Implement logic to process incoming AnnotatedEMPacks, execute the corresponding workflows and store the resulting AnnotatedEMPacks if required.
+- Implement service start-up logic to derive missing schemas for workflow routes with empty output schemas.
+- Implement MongoDB collections to store AnnotatedEMPacks, TransformationWorkflows and WorkflowRoutes.
+
+#### Collections
+
+`AnnotatedEMPackCollection`
+
+Purpose: store incoming and derived annotated EM datapacks that are to be processed or published.
+
+data structure:
+```python
+class AnnotatedEMPack(BaseModel):
+   workflow_route_name: str
```

Comment:
We need something more general here, I think tying the concepts of the AEM with the concept of routing transformations isn't ideal. I would rather go for something like "type" or "schema".